Track file not updated

winterthediplomat opened the pull request:
Fixes alfateam123/Miyuki#2
The `:git` parameter in the gemfile for hydride0/yamazaki is needed to test Miyuki on Travis with the needed patches to make it work. It will be removed when @hydride0 will merge hydride0/yamazaki#11

winterthediplomat committed:
Catch exceptions when notifying

I'm testing on a debian server. I didn't install libnotify, as
it requires gnome (`dnw`).

Tests have to notify things, and the lack of libnotify
raises exceptions that break automated tests on my platform
(when executed with `rake test`).

winterthediplomat committed:
Merge branch 'not-force-libnotify' into serverize

winterthediplomat committed:
Merge branch 'serverize' into track-file-not-updated

winterthediplomat committed:
fix tracking file issue

winterthediplomat committed:
remember to clean up after your tests!

winterthediplomat committed:
rewrite the file check test

winterthediplomat committed:
restore initial refresh time of miyuki.conf

RoxasShadow commented:
Thanks mate. I'll wait for hydride0/yamazaki#11 be merged so we can rid off those references.

RoxasShadow commented on lib/miyuki/version.rb:
```diff
@@ -13,5 +13,5 @@
 
 module Miyuki
-  VERSION = '0.5.8.1'
+  VERSION = '0.5.8.2'
```
This is ok, but usually is good practice leaving to the manteiner the bump of the version :dancer:

alfateam123 replied:
Oh :///:

RoxasShadow commented on lib/miyuki/notifier.rb:
```diff
@@ -19,7 +19,12 @@ def initialize(notifier = nil)
+        @notifier.notify(title, message) if has_notifier?
+      rescue NoMethodError
+        #libnotify is not installed, maybe.
+        #someone may not want to install gnome on its server :V
```
That someone can just turn off notifications tho.

alfateam123 replied:
May I remove notifications from test' configurations? If we need to test them, we need them turned on, and `rake test` will die if libnotify (on servers, for example) is not installed.

RoxasShadow replied:
Sure you can.

alfateam123 replied:
I'd prefer not to touch them, I'll just do a `.patch` file to disable notifications when needed (manual operation, you know what you're doing). It's another file to maintain, but I don't think it's a problem, as they can be generated (and applied) with just a command.

Still, the problem remains: if we implement the test steps related to notifications ([this](https://github.com/RoxasShadow/Miyuki/blob/master/features/tracker.feature#L29) and [that](https://github.com/RoxasShadow/Miyuki/blob/master/features/load_config.feature#L18) ), they have to be executed somehow to make the tests pass.

RoxasShadow replied:
Those tests related to notifications are not implemented yet since I need to find a way to perform them. I suggest you to turn off the notifications for now (yes, removing also `begin ... rescue`).

winterthediplomat committed:
update gemfile after merge of hydride0/yamazaki#11

winterthediplomat committed:
Remove begin ... rescue block from notifications

the `Notifier#notify` call was wrapped in a `begin ... rescue`
block in order to fix tests.

In case you launch tests on a platform where `libnotify` is not
installed, you can apply the patch added with this commit.
Refer to the new `TROUBLESHOOTING.md` file to find fixes
in case you get errors.

RoxasShadow commented:
:+1:
